models/nxos/interfaces/nxos_interfaces.yml

```diff
@@ -0,0 +1,99 @@
+generator_version: 1.0
+metadata:
+  vesion: 1.1
+  status:
+  - preview
+  supported_by:
+  - 'Ansible Network'
+  copyright_str: Copyright 2019 Red Hat
+  license: gpl-3.0.txt
+info:
+  network_os: nxos
+  resource: interfaces
+  version_added: 2.9
+  short_description: 'Manages Interfaces attributes of NX-OS Interfaces.'
+  description:
+  - Manages Interfaces attributes of NX-OS Interfaces.
+  authors:
+  - Trishna Guha (@trishnaguha)
+  extends_documentation_fragment: nxos
+  notes:
+    - Tested against <network_os> 7.3.(0)D1(1) on VIRL
+schema:
+  type: object
+  description: The schema used for the argspec and docstring
+  properties:
+    config:
+      type: array
+      description: The provided configuration
+      items:
+        $ref: '#/definitions/config_dict'
+    state:
+      $ref: '#/definitions/state'
+
+definitions:
+  config_dict:
+    description:
+    - The some_dict.
+    type: object
+    properties:
+      name:
+        description:
+        - Full name of interface, i.e. Ethernet1/1, port-channel10.
+        type: str
+        required: true
+      description:
+        description:
+        - Interface description.
+        type: str
+      enable:
+        description:
+        - Administrative State of interface.
+        type: bool
+        default: true
+      speed:
+        description:
+        - Interface link speed. Applicable for ethernet interface only.
```

Comment:
```suggestion
        - Interface link speed. Applicable for Ethernet interfaces only.
```

Reply:
are there a fixed set of speeds this can be set to? other models have something like:
       type: str
        enum: ['auto', '10', '100', '1000', '2500', '10000']